MekHQ/src/mekhq/campaign/mission/CrewSkillUpgrader.java

```diff
@@ -216,9 +221,18 @@ private boolean extraEligibilityCheck(SpecialAbility spa, Entity entity) {
      * @param entity The entity being manipulated
      * @return Weapon name
      */
-    private String pickRandomWeapon(Entity entity) {
-        int weaponIndex = Compute.randomInt(entity.getIndividualWeaponList().size());
-        return entity.getIndividualWeaponList().get(weaponIndex).getName();
+    private String pickRandomWeapon(Entity entity, boolean clusterOnly) {
+        List<Mounted> weapons = entity.getIndividualWeaponList();
+        List<Mounted> eligibleWeapons = new ArrayList<>();
+        
+        for(Mounted weapon : weapons) {
+            if(SpecialAbility.isWeaponEligibleForSPA(weapon.getType(), Person.T_NONE, clusterOnly)) {
+                eligibleWeapons.add(weapon);
+            }
+        }
+        
+        int weaponIndex = Compute.randomInt(eligibleWeapons.size());
+        return eligibleWeapons.get(weaponIndex).getName();
```

Comment:
What happens if the entity doesn't have any eligible weapons?

Reply:
NPE, probably. I'll add some more defensive coding when I get back to it tonight.

Reply:
Fixed. I suppose I could have it re-roll in that case, but it's a lot more work and I don't think anybody's going to complain about bot unit crews having a tiny amount less SPAs.

Reply:
Does this have the effect of giving the crew the SPA, but making it ineffective because it doesn't match any weapon?

Reply:
Basically. I suppose a better solution would be to have the calling function check whether this method returned anything and avoid adding the SPA in that case.

Reply:
It's functional as is, but probably clearer without the non-functional SPA.